File: haskell/Locations/HallBasement.hs
module Locations.HallBasement where

import Data.Map (Map)
import qualified Data.Map as Map

import Datatypes

-- ============= Objects =============

rat_poison :: ItemMap
rat_poison = ItemMap
  { itemName = "Rat Poison"
  , itemDescription = ["Little red pellets which are used for deratization."]
  , itemEat =
      [ "You've eaten all the rat poison you could see in the hallway."
      , "You feel sick you die a moment later."
      , "Did you really think that it could end in any other way?"
      ]
  , itemEatDeath = []
  , itemEatDeathChance = 100
  }

-- ============= People =============

-- ============= Location ===========

hall_basement :: LocationMap
hall_basement = LocationMap
  { locationDescription =
      [ "You look around and you see entrance to the laundry room."
      , "You can also see rat poison in the corners of the hallway"
      ]
  , locationItems =
      Map.fromList
        [ ("rat_poison", rat_poison)
        ]
  , locationPeople = Map.empty
  , locationPaths =
      ["hall_ground_floor", "laundry_room"]
  }

File: prolog/locations/hall_basement.pl
:- multifile die/0.

% ============== Info ==============

info(hall_basement) :-
  write('You look around and you see entrance to the laundry room.'), nl,
  write('You can also see rat poison in the corners of the hallway'), nl,
  !.

% ============= Objects =============
at(rat_poison, hall_basement).
do_describe(rat_poison) :-
  write('Little red pellets which are used for deratization.'), nl,
  !.
do_eat(rat_poison) :-
  write('You''ve eaten all the rat poison you could see in the hallway.'), nl,
  write('You feel sick you die a moment later.'), nl,
  write('Did you really think that it could end in any other way?'), nl,
  die,
  !.

% ============= People =============

% ============= Possible goto Locations =============

path(hall_ground_floor, hall_basement).
path(laundry_room, hall_basement).
